vm/vm.go

```diff
@@ -159,13 +168,46 @@ func (vm *VM) Initialize(
 	vm.networkManager = network.NewManager(vm.snowCtx.Log, vm.snowCtx.NodeID, appSender)
 
 	vm.baseDB = baseDB
+	pebbleConfig := pebble.NewDefaultConfig()
+
+	vm.vmDB, err = storage.New(pebbleConfig, vm.snowCtx.ChainDataDir, blockDB, vm.snowCtx.Metrics)
+	if err != nil {
+		return err
+	}
+
+	vm.rawStateDB, err = storage.New(pebbleConfig, vm.snowCtx.ChainDataDir, stateDB, vm.snowCtx.Metrics)
+	if err != nil {
+		return err
+	}
+
+	// TODO do not expose entire context to the Controller
+	//
+	// Note: does not copy the consensus lock but this is safe because the
+	// consensus lock does not work over rpc anyways
+	controllerContext := &snow.Context{
+		NetworkID:      vm.snowCtx.NetworkID,
+		SubnetID:       vm.snowCtx.SubnetID,
+		ChainID:        vm.snowCtx.ChainID,
+		NodeID:         vm.snowCtx.NodeID,
+		PublicKey:      vm.snowCtx.PublicKey,
+		XChainID:       vm.snowCtx.XChainID,
+		CChainID:       vm.snowCtx.CChainID,
+		AVAXAssetID:    vm.snowCtx.AVAXAssetID,
+		Log:            vm.snowCtx.Log,
+		Keystore:       vm.snowCtx.Keystore,
+		SharedMemory:   vm.snowCtx.SharedMemory,
+		BCLookup:       vm.snowCtx.BCLookup,
+		Metrics:        vm.snowCtx.Metrics,
+		WarpSigner:     vm.snowCtx.WarpSigner,
+		ValidatorState: vm.snowCtx.ValidatorState,
+		ChainDataDir:   filepath.Join(vm.snowCtx.ChainDataDir, vmDataDir),
+	}
```

Comment:
(not necessary for this PR) thoughts on adding one more layer of nesting here ie.

outerVM
- block
- state

innerVM
- vm

Reply:
Don't really have thoughts on this... if someone has strong opinions I can be convinced for or against this though...